src/arviz_plots/plots/ppcplot.py

```diff
@@ -0,0 +1,385 @@
+    observed_distribution = distribution.sel(model="observed_data")
+    print(
+        f"\nobserved distri = {observed_distribution.obs!r}"
+    )  # the obs data was auto broadcasted all over the chain and draw dims
+    # but we can ignore the other dims and just subset it to 1 chain and 1 draw and then use the
+    # resulting subsetted variable data
+    observed_distribution = observed_distribution.sel(chain=0, draw=0)
+    print(f"\nobserved distri = {observed_distribution.obs!r}")
+
+    # density calculation for observed variables
+    density_kwargs = copy(plot_kwargs.get(kind, {}))
+
+    if density_kwargs is not False:
+        density_dims, _, density_ignore = filter_aes(plot_collection, aes_map, "kde", sample_dims)
+        print(f"\ndensity_dims = {density_dims}\ndensity_ignore= {density_ignore}")
+
+    for dim in flatten:  # flatten is the list of user defined dims to flatten or all dims
+        obs_density_dims = {dim}.union(
+            density_dims
+        )  # dims to be reduced now includes the flatten ones and not just sample dims
+
+    obs_density = observed_distribution.azstats.kde(
+        dims=obs_density_dims, **stats_kwargs.get("density", {})
+    )
+    print(f"\nobserved data density = {obs_density}")
+
+    plot_collection.map(
+        line_xy, "kde", data=obs_density, ignore_aes=density_ignore, **density_kwargs
+    )
+
+    # ---------STEP 2 (PPC data)-------------
+    # subset distribution for predictive data model, reshape and pick samples and flatten and
+    # then call kde to get density info then call map() again with this density info
+    # (algorithm of legacy plot ppc followed for this, but implemented in refactored way)
+    predictive_distribution = distribution.sel(model="posterior_predictive")
+    print(f"\nposterior predictive distri = {predictive_distribution.obs!r}")
+
+    predictive_distribution = predictive_distribution.stack(ppc_dim=("chain", "draw"))  # reshaping
+    predictive_distribution = predictive_distribution.assign_coords(
+        ppc_dim=np.arange(predictive_distribution.sizes["ppc_dim"])
+    )
+    print(f"\nposterior predictive distri reshaped = {predictive_distribution!r}")
+    print(f"\nposterior predictive distri stacked variable = {predictive_distribution.obs!r}")
+
+    # selecting sampled values from predictive_distribution only
+    predictive_distribution = predictive_distribution.isel(ppc_dim=pp_sample_ix)
+    print(f"\nposterior predictive distri selected samples = {predictive_distribution.obs!r}")
+
+    for dim in flatten_pp:  # flatten is the list of user defined dims to flatten or all dims
+        pp_density_dims = {dim}.union(
+            density_dims
+        )  # dims to be reduced now includes the flatten ones and not just sample dims
+    # ppc_dim should not be flattened- this is because subselections of this correspond to
+    # pp samples
+
+    pp_densities = []
+    pp_xs = []
+
+    for i in range(predictive_distribution.sizes["ppc_dim"]):
```

Comment:
So this is sort the approach that should be taken when calling map on plot_ppc as well, right - as in sample_dims should be all of the non chain and draw dimensions (by default, though a user could select less) since these are the dims being flattened and used to generate KDEs, while the actual pp sample dims chain and draw are set as "overlay" in the aesthetic mapping?

Reply:
In the example `sample_dims` has nothing to do with `.map`, the need for `sample_dims="school"` is because `sample_dims` are the dimensions being reduced by `plot_dist` when computing the kde.

In plot_ppc, we split the dimensions into 3 non-overlapping groups:
* overlay dimensions -> here `sample_dims`
* facet dimensions -> what used to be `flatten` argument/now modifiable also via pc_kwargs
* rest/leftover/reduce (defined by elimination) -> these are the ones that should be reduced when computing the kde

In this particular example, they are respectively `[chain, draw]`, `[]` and `[school]`

Reply:
Got it, this way of splitting makes sense. I've modified the code in my latest commit to take into account these three kinds of dimensions (for overlay/facetting/reducing)